File: src_backup_phpstan/Entity/Offres/Commentaire.php
<?php

namespace App\Entity\Offres;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'commentaire')]
class Commentaire
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: 'id_commentaire', type: 'integer')]
    private ?int $idCommentaire = null;

    #[ORM\Column(name: 'id_offre', type: 'integer')]
    private ?int $idOffre = null;

    #[ORM\Column(name: 'id_utilisateur', type: 'integer')]
    private ?int $idUtilisateur = null;

    #[ORM\Column(name: 'contenu', type: 'text')]
    private ?string $contenu = null;

    #[ORM\Column(name: 'date_commentaire', type: 'datetime', nullable: true)]
    private ?\DateTimeInterface $dateCommentaire = null;

    public function getIdCommentaire(): ?int
    {
        return $this->idCommentaire;
    }

    public function getIdOffre(): ?int
    {
        return $this->idOffre;
    }

    public function setIdOffre(?int $idOffre): self
    {
        $this->idOffre = $idOffre;
        return $this;
    }

    public function getIdUtilisateur(): ?int
    {
        return $this->idUtilisateur;
    }

    public function setIdUtilisateur(?int $idUtilisateur): self
    {
        $this->idUtilisateur = $idUtilisateur;
        return $this;
    }

    public function getContenu(): ?string
    {
        return $this->contenu;
    }

    public function setContenu(?string $contenu): self
    {
        $this->contenu = $contenu;
        return $this;
    }

    public function getDateCommentaire(): ?\DateTimeInterface
    {
        return $this->dateCommentaire;
    }

    public function setDateCommentaire(?\DateTimeInterface $dateCommentaire): self
    {
        $this->dateCommentaire = $dateCommentaire;
        return $this;
    }
}

File: src/main/java/org/model/Offres/Commentaire.java
package org.model.Offres;

import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class Commentaire {

    private int           idCommentaire;
    private int           idOffre;
    private int           idUtilisateur;
    private String        contenu;
    private LocalDateTime dateCommentaire;

    // Dénormalisé pour affichage
    private String        nomAuteur;
    private String        prenomAuteur;

    private static final DateTimeFormatter FMT = DateTimeFormatter.ofPattern("dd/MM/yyyy HH:mm");

    public Commentaire() {}

    // ── Getters / Setters ──────────────────────────────────────
    public int           getIdCommentaire()              { return idCommentaire; }
    public void          setIdCommentaire(int v)         { this.idCommentaire = v; }

    public int           getIdOffre()                    { return idOffre; }
    public void          setIdOffre(int v)               { this.idOffre = v; }

    public int           getIdUtilisateur()              { return idUtilisateur; }
    public void          setIdUtilisateur(int v)         { this.idUtilisateur = v; }

    public String        getContenu()                    { return contenu; }
    public void          setContenu(String v)            { this.contenu = v; }

    public LocalDateTime getDateCommentaire()            { return dateCommentaire; }
    public void          setDateCommentaire(LocalDateTime v){ this.dateCommentaire = v; }

    public String        getNomAuteur()                  { return nomAuteur; }
    public void          setNomAuteur(String v)          { this.nomAuteur = v; }

    public String        getPrenomAuteur()               { return prenomAuteur; }
    public void          setPrenomAuteur(String v)       { this.prenomAuteur = v; }

    // ── Helper ─────────────────────────────────────────────────
    public String getAuteurLabel() {
        String n = nomAuteur    != null ? nomAuteur    : "";
        String p = prenomAuteur != null ? prenomAuteur : "";
        return (n + " " + p).trim();
    }

    public String getDateLabel() {
        return dateCommentaire != null ? dateCommentaire.format(FMT) : "—";
    }
}